lib/facility_factory.rb

```diff
@@ -0,0 +1,8 @@
+class FacilityFactory
+
+  def create_facilities(facility_data)
+    facility = facility_data.map do |facility|
+      Facility.new(facility)
+    end
+  end
```

Comment:
This works. You could also put the logic for checking all the keys within the data into this class, you put it into the Facility class's initialize method. Either is fine.